src/main/java/dan200/computercraft/shared/peripheral/speaker/SpeakerPeripheral.java

```diff
@@ -55,6 +55,11 @@ public BlockPos getPos()
         return m_speaker.getPos();
     }
 
+    public synchronized boolean madeSound()
+    {
+        return (m_clock - m_lastPlayTime <= 20) ;
```

Comment:
You really love your functional programing don't you?
But i agree it would make sense to allow for some interesting interaction at later changes.